docs/src/index.md

```diff
@@ -183,57 +221,65 @@ Similarly every `pullback` returns an extra `∂self`, which for things without
 
 ```julia
-y, f_pushforward = frule(f, a, b, c)
-ẏ = f_pushforward(1, 1, 1, 1)  # for appropriate `1`-like perturbation.
+direction = (1.5, 0.4, -1) # (ȧ, ḃ, ċ)
+y, ẏ = frule(f, a, b, c, Zero(), 1, 0, 0)
 ```
 
-Then we have that `ẏ` is the _total derivative_ of `f` at `(a, b, c)`, written mathematically as ``df_{(a,b,c)}``
+On the basis directions one gets the partial derivatives of `y`:
+```julia
+y, dy_da = frule(f, a, b, c, Zero(), 1, 0, 0)
+y, dy_db = frule(f, a, b, c, Zero(), 0, 1, 0)
+y, dy_dc = frule(f, a, b, c, Zero(), 0, 0, 1)
+```
 
 Similarly, the most trivial use of `rrule` and returned `pullback` is to calculate the [Gradient](https://en.wikipedia.org/wiki/Gradient):
 
 ```julia
 y, f_pullback = rrule(f, a, b, c)
 ∇f = f_pullback(1)  # for appropriate `1`-like seed.
-s̄elf, ā, b̄, c̄ = ∇f
+s̄elf, ā, b̄, c̄ = ∇f
 ```
 Then we have that `∇f` is the _gradient_ of `f` at `(a, b, c)`.
 And we thus have the partial derivatives ``\overline{\mathrm{self}}, = \dfrac{∂f}{∂\mathrm{self}}``, ``\overline{a} = \dfrac{∂f}{∂a}``, ``\overline{b} = \dfrac{∂f}{∂b}``, ``\overline{c} = \dfrac{∂f}{∂c}``, including the and the self-partial derivative, ``\overline{\mathrm{self}}``.
 
 ### Differentials
 
-The values that come back from pullbacks or pushforwards are not always the same type as the input/outputs of the original function.
-They are differentials, which correspond roughly to something able to represent the difference between two values of the original types.
+The values that come back from pullbacks or pushforwards are not always the same type as the input/outputs of the primal function.
+They are differentials, which correspond roughly to something able to represent the difference between two values of the primal types.
 A differential might be such a regular type, like a `Number`, or a `Matrix`, matching to the original type;
 or it might be one of the `AbstractDifferential` subtypes.
 
 Differentials support a number of operations.
 Most importantly: `+` and `*`, which let them act as mathematical objects.
-And `extern` which converts `AbstractDifferential` types into a conventional non-ChainRules type.
 
 The most important `AbstractDifferential`s when getting started are the ones about avoiding work:
 
- - `Thunk`: this is a deferred computation. A thunk is a [word for a zero argument closure](https://en.wikipedia.org/wiki/Thunk). A computation wrapped in a `@thunk` doesn't get evaluated until `extern` is called on the `Thunk`. More on thunks later.
+ - `Thunk`: this is a deferred computation. A thunk is a [word for a zero argument closure](https://en.wikipedia.org/wiki/Thunk). A computation wrapped in a `@thunk` doesn't get evaluated until `unthunk` is called on the `Thunk`. (`unthunk` is a no-op on nonthunked inputs.)
  - `One`, `Zero`: There are special representations of `1` and `0`. They do great things around avoiding expanding `Thunks` in multiplication and (for `Zero`) addition.
```

Comment:
```suggestion
 - `Zero`: There is a special representation of `0`. It's defined as the differential that leaves and differential that it's added to unchanged. It's great for avoiding computation.
```
I just don't really want to talk about `One` in the docs since it's looking increasingly likely that we'll drop it in the near future.

Reply:
i don't feel strongly, but i'd slightly leave change to `One` in docs to be included in (or after) any changes to `One` itself

Reply:
Yeah, you're probably right. I'm being overzealous.